_posts/2024-08-24-GSoC2024-dingo.md

```diff
@@ -0,0 +1,119 @@
+
+A `plot_corr_matrix` function to visualize the correlation matrix as a heatmap plot was implemented too.
+
+This figure illustrates a heatmap from a symmetrical correlation matrix without pearson or indicator filtering:
+<center>
+    <img src="https://sotiristouliopoulos.github.io/dingo/img/corr_matrix_no_cutoffs.png"><br><br>
+</center>
+
+This figure illustrates a heatmap from a triangular correlation matrix with both pearson and indicator filtering:
+<center>
+    <img src="https://sotiristouliopoulos.github.io/dingo/img/corr_matrix_pearson_indicator_triangle.png"><br><br>
+</center>
+
+
+## Clustering
+
+- Clustering based on a dissimilarity matrix reveals groups of reactions with similar correlation values.
+- Reactions within the same cluster may contribute to the same pathways.
+
+A `cluster_corr_reactions` function that hierarchically clusters a correlation matrix 
+alongside a `plot_dendrogram` function that plots a dendrogram were implemented.
+
+This figure illustrates a dendrogram created from a filtered correlation matrix:
+<center>
+    <img src="https://sotiristouliopoulos.github.io/dingo/img/dendrogram_pearson.png"><br><br>
+</center>
+
+Distinct clusters are observed. Graphs will reveal if these clusters interact with other clusters or reactions.
+
+
+## Graphs
+
+- Graphs creation can reveal networks of correlated reactions, potentially corresponding to metabolic pathways.
+
+A `graph_corr_matrix` function that creates graphs from a correlation matrix 
+alongside a `plot_graph` function that plots the graphs were implemented.
+
+This figure illustrates a graph created from a correlation matrix without pearson or indicator filtering:
+<center>
+    <img src="https://sotiristouliopoulos.github.io/dingo/img/graph_no_cutoffs.png"><br><br>
+</center>
+
+This figure illustrates a subgraph created from a filtered correlation matrix:
+<center>
+    <img src="https://sotiristouliopoulos.github.io/dingo/img/subgraph_pearson.png"><br><br>
+</center>
+
+This subgraph has 9 nodes that correspond to 9 reactions close to each other in the topology of the E. coli core model. These reactions are: `PGI, G6PDH2R, PGL, GND, RPE, RPI, TKT1, TALA, TKT2`. 
+
+The topology of these reactions can be seen in the figure below from `ESCHER`:
+<center>
+    <img src="https://sotiristouliopoulos.github.io/dingo/img/escher_subgraph.png"><br><br>
+</center>
+
+We observe that `PGI` seems to contribute to a different pathway. However it shares a common metabolite with `G6PDH2r`. 
+If we apply a stricter pearson cutoff (e.g., 0.99999), this reaction is removed from this subgraph, leaving only the remaining reactions. This is an important observation: looser cutoffs lead to wider sets of connected reactions, forming larger metabolic pathways.
+
```

Comment:
I think a small conclusion section would be useful.

Reply:
That's important, yes. 
What do we gain from each of these features? 
For example, why do we care to reduce the model in the first place? 
And of course, what do we learn from the graph.